diff --git a/CMakeLists.txt b/CMakeLists.txt
index 481393b..d5e1002 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -350,7 +350,12 @@ endif()
 if(CPPTRACE_GET_SYMBOLS_WITH_LIBDWARF)
   target_compile_definitions(${target_name} PUBLIC CPPTRACE_GET_SYMBOLS_WITH_LIBDWARF)
   if(CPPTRACE_USE_EXTERNAL_LIBDWARF)
-    find_package(libdwarf REQUIRED)
+    if(NOT CPPTRACE_FIND_LIBDWARF_WITH_PKGCONFIG)
+      find_package(libdwarf REQUIRED)
+    else()
+      find_package(PkgConfig)
+      pkg_check_modules(LIBDWARF REQUIRED libdwarf)
+    endif()
   else()
     include(FetchContent)
     # First, dependencies: Zstd and zlib (currently relying on system zlib)
diff --git a/cmake/OptionVariables.cmake b/cmake/OptionVariables.cmake
index 5589a8d..d432a3c 100644
--- a/cmake/OptionVariables.cmake
+++ b/cmake/OptionVariables.cmake
@@ -159,9 +159,11 @@ if(PROJECT_IS_TOP_LEVEL)
 endif()

 option(CPPTRACE_USE_EXTERNAL_LIBDWARF "" OFF)
+option(CPPTRACE_FIND_LIBDWARF_WITH_PKGCONFIG "" OFF)
 option(CPPTRACE_USE_EXTERNAL_ZSTD "" OFF)
 option(CPPTRACE_CONAN "" OFF)
 option(CPPTRACE_VCPKG "" OFF)
 option(CPPTRACE_SANITIZER_BUILD "" OFF)
 option(CPPTRACE_WERROR_BUILD "" OFF)
 option(CPPTRACE_POSITION_INDEPENDENT_CODE "" ON)
diff --git a/cmake/in/cpptrace-config-cmake.in b/cmake/in/cpptrace-config-cmake.in
index fc017ad..06ab746 100644
--- a/cmake/in/cpptrace-config-cmake.in
+++ b/cmake/in/cpptrace-config-cmake.in
@@ -14,7 +14,9 @@ if(@CPPTRACE_GET_SYMBOLS_WITH_LIBDWARF@)
     set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH_OLD}")
     unset(CMAKE_MODULE_PATH_OLD)
   endif()
-  find_dependency(libdwarf REQUIRED)
+  if(NOT @CPPTRACE_FIND_LIBDWARF_WITH_PKGCONFIG@)
+    find_dependency(libdwarf REQUIRED)
+  endif()
 endif()

 # We cannot modify an existing IMPORT target
